docs/code/MlNetCookBook.md

```diff
@@ -0,0 +1,707 @@
+// This will give the entire dataset: make sure to only take several row
+// in case the dataset is huge.
+var featureColumns = transformedData.GetColumn(r => r.AllFeatures)
+    .Take(20).ToArray();
+
+// The same extension method also applies to the dynamic-typed data, except you have to
+// specify the column name and type:
+var dynamicData = transformedData.AsDynamic;
+var sameFeatureColumns = dynamicData.GetColumn<string[]>(env, "AllFeatures")
+    .Take(20).ToArray();
+```
+
+The above code assumes that we defined our `InspectedRow` class as follows:
+```c#
+private class InspectedRow
+{
+    public bool IsOver50K;
+    public string Workclass;
+    public string Education;
+    public string MaritalStatus;
+    public string[] AllFeatures;
+}
+```
+
+## How do I train a regression model?
+
+Generally, in order to train any model in ML.NET, you will go through three steps:
+1. Figure out how the training data gets into ML.NET in a form of an `IDataView`
+2. Build the 'learning pipeline' as a sequence of elementary 'operators' (estimators).
+3. Call `Fit` on the pipeline to obtain the trained model.
+
+Example file (https://github.com/dotnet/machinelearning/blob/master/test/data/generated_regression_dataset.csv):
+```
+feature_0;feature_1;feature_2;feature_3;feature_4;feature_5;feature_6;feature_7;feature_8;feature_9;feature_10;target
+-2.75;0.77;-0.61;0.14;1.39;0.38;-0.53;-0.50;-2.13;-0.39;0.46;140.66
+-0.61;-0.37;-0.12;0.55;-1.00;0.84;-0.02;1.30;-0.24;-0.50;-2.12;148.12
+-0.85;-0.91;1.81;0.02;-0.78;-1.41;-1.09;-0.65;0.90;-0.37;-0.22;402.20
+```
+
+In the file above, the last column (12th) is label that we predict, and all the preceding ones are features.
+
+```c#
+// Create a new environment for ML.NET operations. It can be used for exception tracking and logging, 
+// as well as the source of randomness.
+var env = new LocalEnvironment();
+
+// Step one: read the data as an IDataView.
+// First, we define the reader: specify the data columns and where to find them in the text file.
+var reader = TextLoader.CreateReader(env, ctx => (
+        // We read the first 11 values as a single float vector.
+        FeatureVector: ctx.LoadFloat(0, 10),
+        // Separately, read the target variable.
+        Target: ctx.LoadFloat(11)
+    ),
+    // The data file has header.
+    hasHeader: true,
+    // Default separator is tab, but we need a semicolon.
+    separator: ';');
+
+
+// Now read the file (remember though, readers are lazy, so the actual reading will happen when the data is accessed).
+var trainData = reader.Read(new MultiFileSource(trainDataPath));
+
+// Step two: define the learning pipeline. 
+// We know that this is a regression task, so we create a regression context: it will give us the algorithms
+// we need, as well as the evaluation procedure.
+var regression = new RegressionContext(env);
+
+// We 'start' the pipeline with the output of the reader.
+var learningPipeline = reader.MakeNewEstimator()
```

Comment:
>MakeNewEstimator [](start = 30, length = 16)

There are a couple comments here we may want to double click on, since I've noticed them sometimes being sources of confusion.

I've a couple cases where people get confused that this `reader` object can actually be used to read multiple files.

The idea that this `MakeNewEstimator` is the start of a new chain is one thing you personally believed was confusing, so I had expected you to expand a bit to forestall that confusion.